### DISCUSSION TITLE
Anyone here implemented Theme change successfully with Voyager

### DISCUSSION THREAD

**Discussion by RaheemJnr** [adrielcafe/voyager#478] [General]
I have been trying to implement theme change in 2 of my app but it being a hassle after extensive debug period I figured the common thing between them is that both app uses Voyager because they are a multiplatform app, I figured after a successful theme change and the state changes Voyager doesn't recompose to reflect the current (newly) selected theme until I close the app and open again, so my question now is had anyone faced similar issue and what advice will you give me on how to overcome it or probably a code example.

**Comment by AbanobNageh** [comment]
@RaheemJnr
Hi, I have a simple Android app created with Compose and Voyager [here](https://github.com/AbanobNageh/RecipeApp). I have implemented the theme change feature successfully. I am not sure if this is the best example since you mentioned that your apps are multiplatform apps but I hope it can help you in some way.

**Reply by RaheemJnr** [reply]
Thanks i'll give it a trial

**Reply by RaheemJnr** [reply]
Thanks for your help but after lots of debugging I discovered the problem was with **TabNavigator**  everything else reacted to the theme change except the view within **TabNavigator** which is the majority of the screen @Syer10 any help with this??

**Comment by RaheemJnr** [comment]
Thank you so much! It finally worked after downgrading my voyager version.
